src/Changelog.js

```diff
@@ -142,74 +183,122 @@ export default class Changelog {
     return Object.keys(committers).map(k => committers[k]).sort();
   }
 
-  getCommitInfo() {
+  getCommitsInfo() {
     const commits = this.getListOfCommits();
 
     progressBar.init(commits.length);
 
-    var logs = commits.map(commit => {
+    const commitsInfo = commits.map(commit => {
+      // commit is formatted as following:
+      // <short-hash>;<ref-name>;<summary>;<date>
+      const parts = commit.split(";");
+      const sha = parts[0];
+      const _refs = parts[1].split(",");
+      let tag;
+      if (_refs.length === 1) {
+        // "tag: <tag-name>
+        tag = _refs[0].replace(/tag:\s(.*?)$/, "$1").trim();
+      } else if (_refs.length === 4) {
+        // "HEAD -> master, tag: <tag-name>, origin/master, origin/HEAD"
+        tag = _refs[1].replace(/tag:\s(.*?)$/, "$1").trim();
+      }
+      const message = parts[2];
+      const date = parts[3];
 
-      var sha = commit.slice(0, 7);
-      var message = commit.slice(8);
-      var response;
       progressBar.tick(sha);
 
-      var mergeCommit = message.match(/\(#(\d+)\)$/);
+      const mergeCommit = message.match(/\(#(\d+)\)$/);
 
-      if (message.indexOf("Merge pull request ") === 0) {
-        var start = message.indexOf("#") + 1;
-        var end = message.slice(start).indexOf(" ");
-        var issueNumber = message.slice(start, start + end);
+      const commitInfo = {
+        commitSHA: sha,
+        message: message,
+        // Note: Only merge commits or commits referencing an issue / PR
+        // will be kept in the changelog.
+        labels: [],
+        tag,
+        date
+      };
 
-        response = this.remote.getIssueData(issueNumber);
-        response.commitSHA = sha;
-        response.mergeMessage = message;
-        return response;
-      } else if (mergeCommit) {
-        var issueNumber = mergeCommit[1];
-        response = this.remote.getIssueData(issueNumber);
+      if (message.indexOf("Merge pull request ") === 0 || mergeCommit) {
+        let issueNumber;
+        if (message.indexOf("Merge pull request ") === 0) {
+          const start = message.indexOf("#") + 1;
+          const end = message.slice(start).indexOf(" ");
+          issueNumber = message.slice(start, start + end);
+        } else
+          issueNumber = mergeCommit[1];
+
+        const response = this.remote.getIssueData(issueNumber);
         response.commitSHA = sha;
         response.mergeMessage = message;
-        return response;
+        Object.assign(commitInfo, response);
```

Comment:
Btw, do you want to automate the release process (e.g. semantic-release) or still keep it manual (`npm publish`)? At least I would consider having a script to bump the version, tag it and publish it.
But I'm not sure how you guys are used to work and prefer to do this sorts of stuff, so yeah just asking... :)